test/net_components/layers/conv2d.jl

```diff
@@ -76,953 +170,333 @@ using MIPVerify: check_size, increment!
-            input_size = (1, 5, 6, 1)
-            input = reshape([1:prod(input_size);], input_size)
-            filter_size = (3, 3, 1, 1)
-            filter = ones(filter_size...)
-            bias_size = (1, )
-            bias = [0]
-            stride = 1
-            true_output_raw = [
+            expected_output_2d = [
                 63  72  81;
                 108 117 126;
                 153 162 171;
                 198 207 216
             ]
-            true_output = reshape(transpose(true_output_raw), (1, 3, 4, 1))
-            p = Conv2d(filter, bias, stride, valid)
-            @testset "Numerical Input, Numerical Layer Parameters" begin
-                evaluated_output = MIPVerify.conv2d(input, p)
-                @test evaluated_output == true_output
-            end
-            @testset "Numerical Input, Variable Layer Parameters" begin
-                m = TestHelpers.get_new_model()
-                filter_v = map(_ -> @variable(m), CartesianIndices(filter_size))
-                bias_v = map(_ -> @variable(m), CartesianIndices(bias_size))
-                p_v = Conv2d(filter_v, bias_v, stride, valid)
-                output_v = MIPVerify.conv2d(input, p_v)
-                @constraint(m, output_v .== true_output)
-                solve(m)
-
-                p_solve = MIPVerify.Conv2d(getvalue(filter_v), getvalue(bias_v), stride, valid)
-                solve_output = MIPVerify.conv2d(input, p_solve)
-                @test solve_output≈true_output
-            end
-            @testset "Variable Input, Numerical Layer Parameters" begin
-                m = TestHelpers.get_new_model()
-                input_v = map(_ -> @variable(m), CartesianIndices(input_size))
-                output_v = MIPVerify.conv2d(input_v, p)
-                @constraint(m, output_v .== true_output)
-                solve(m)
-
-                solve_output = MIPVerify.conv2d(getvalue(input_v), p)
-                @test solve_output≈true_output
-            end
-        end
-
-        @testset "conv2d with 'valid' padding, stride=1, filter width != filter height" begin
-            input_size = (1, 5, 5, 1)
-            input = reshape([1:prod(input_size);], input_size)
-            filter_size = (2, 3, 1, 1)
-            filter = ones(filter_size...)
-            bias_size = (1, )
-            bias = [0]
-            stride = 1
-            true_output_raw = [
-                39  45  51;
-                57  69  75;
-                81  87  99;
-                105 111 117
+            test_convolution_layer_with_default_values(
+                (1, 5, 6, 1),
+                (3, 3, 1, 1),
+                transpose(expected_output_2d),
+                1,
+                ValidPadding()
+            )
+        end
+
+        @testset "conv2d with 'valid' padding, stride = 1, filter width != filter height" begin
+            expected_output_2d = [
```

Comment:
I had to reshape https://github.com/samuelemarro/MIPVerify.jl/pull/3/files#diff-2c30731b2ea2931fab8b970b786c5907L349